e2e/nomostest/config_sync.go

```diff
@@ -816,6 +817,128 @@ func RepoSyncObjectV1Beta1(nn types.NamespacedName, repoURL string, sourceFormat
 	return rs
 }
 
+// UpdateRootSyncSpec updates the RootSync object on the cluster with the provided
+// spec.
+func (nt *NT) UpdateRootSyncSpec(rs *v1beta1.RootSync) error {
+	var clusterRootSync v1beta1.RootSync
+	if err := nt.KubeClient.Get(rs.Name, rs.Namespace, &clusterRootSync); err != nil {
+		return err
+	}
+	clusterRootSync.Spec = rs.Spec
+	return nt.KubeClient.Update(&clusterRootSync)
+}
+
+// UpdateRepoSyncSpec updates the RepoSync object on the cluster with the provided
+// spec.
+func (nt *NT) UpdateRepoSyncSpec(rs *v1beta1.RepoSync) error {
+	var clusterRepoSync v1beta1.RepoSync
+	if err := nt.KubeClient.Get(rs.Name, rs.Namespace, &clusterRepoSync); err != nil {
+		return err
+	}
+	clusterRepoSync.Spec = rs.Spec
+	return nt.KubeClient.Update(&clusterRepoSync)
+}
+
+// RootSyncObjectOCI returns a RootSync object that syncs the provided OCIImage.
+func (nt *NT) RootSyncObjectOCI(name string, image *registryproviders.OCIImage) *v1beta1.RootSync {
+	rs := RootSyncObjectV1Beta1FromRootRepo(nt, name)
+	rs.Spec.SourceType = string(v1beta1.OciSource)
+	rs.Spec.Oci = &v1beta1.Oci{
+		Image: image.FloatingBranchTag(),
+		Auth:  configsync.AuthNone,
+	}
+	switch *e2e.OCIProvider {
+	case e2e.Local:
+		// Local provider requires CA cert because host is not well known
+		rs.Spec.Oci.CACertSecretRef = &v1beta1.SecretReference{
+			Name: PublicCertSecretName(RegistrySyncSource),
+		}
+	case e2e.ArtifactRegistry:
+		// AR provider requires auth because the registry is private
+		rs.Spec.Oci.Auth = configsync.AuthGCPServiceAccount
```

Comment:
Done. I'll need to revalidate the tests using this auth method

Reply:
Is it an issue with k8sserviceaccount support in RepoSync or just the test infra configuration?

Reply:
Did you add the right IAM to allow the KSA to use artifactregistry in your project?

- gcpserviceaccount needs you to authorize the GSA. Then the KSA needs IAM to impersonate the GSA.
- k8sserviceaccount just needs you to authorize the KSA directly.

Reply:
I was going off of https://github.com/GoogleContainerTools/kpt-config-sync/pull/1121, which didn't add any new provisioning config/logic for IAM. After inspecting the CI project, I see that new IAM must have been added manually for the root-reconciler KSA. I suggest adding this to the provisioning config so that it is automated in the future

Reply:
https://github.com/GoogleContainerTools/kpt-config-sync/pull/1126

Reply:
Yes, those IAM were manually added. #1127 added the configs.

Reply:
Does it work now?

Reply:
Yes It did when I tested with the RepoSync IAM from https://github.com/GoogleContainerTools/kpt-config-sync/pull/1126.